framework/src/variables/MooseVariableBase.C

```diff
@@ -107,8 +111,8 @@ MooseVariableBase::MooseVariableBase(const InputParameters & parameters)
     paramError("family", "finite volume (fv=true) variables must be have MONOMIAL family");
   if (getParam<bool>("fv") && _fe_type.order != 0)
     paramError("order", "finite volume (fv=true) variables currently support CONST order only");
-
-  if (_count > 1)
+  bool av = getParam<bool>("av");
+  if (av)
```

Comment:
Should add an assertion on the other branch to make sure that the number of components is 1

Also above add a check that if components > 1 and array is set by user and is false to error with a param error, and add a test for this